File: bookmarker-java/src/main/java/com/sivalabs/bookmarker/config/BookmarkerProperties.java
package com.sivalabs.bookmarker.config;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;

@ConfigurationProperties(prefix = "bookmarker")
@Data
public class BookmarkerProperties {
    private boolean importDataEnabled = true;
    private String importFilePath;
    private String adminEmail;
    private String logstashHost;
}

File: backend/src/main/kotlin/com/sivalabs/bookmarker/config/BookmarkerProperties.kt
package com.sivalabs.bookmarker.config

import com.sivalabs.bookmarker.utils.Constants.DEFAULT_JWT_TOKEN_EXPIRES
import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "bookmarker")
class BookmarkerProperties {
    var jwt: JwtConfig = JwtConfig()

    class JwtConfig {
        var issuer: String = "bookmarker"
        var header: String = "Authorization"
        var expiresIn: Long = DEFAULT_JWT_TOKEN_EXPIRES
        var secret: String = ""
    }
}
